src/qdrant_client/conversions/vectors.rs

```diff
@@ -0,0 +1,150 @@
+use std::collections::HashMap;
+
+use crate::qdrant::vectors::VectorsOptions;
+use crate::qdrant::{NamedVectors, SparseIndices, Vector, Vectors};
+
+impl From<Vec<f32>> for Vector {
+    fn from(vector: Vec<f32>) -> Self {
+        Vector::new_dense(vector)
+    }
+}
+
+impl From<Vec<(u32, f32)>> for Vector {
+    fn from(tuples: Vec<(u32, f32)>) -> Self {
+        Self::from(tuples.as_slice())
+    }
+}
+
+impl From<Vec<Vec<f32>>> for Vector {
+    fn from(vectors: Vec<Vec<f32>>) -> Self {
+        Vector::new_multi(vectors)
+    }
+}
+
+// Since we construct two new Vec's anyway it's fine to source from a reference
+impl From<&[(u32, f32)]> for Vector {
+    fn from(tuples: &[(u32, f32)]) -> Self {
+        let mut indices = Vec::with_capacity(tuples.len());
+        let mut values = Vec::with_capacity(tuples.len());
+        for (i, w) in tuples {
+            indices.push(*i);
+            values.push(*w);
+        }
```

Comment:
This does everything for us, including proper capacity allocation:

```suggestion
        let (indices, values) = tuples.iter().cloned().unzip();
```